tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,241 @@
+
+First, we will do this for the low-level "SJER/Reflectance" group. 
+
+```{r ref-attributes}
+
+a=h5readAttributes(f,"/SJER/Reflectance/")
+fid <- H5Fopen(nf)
+g=H5Gopen(fid, "SJER/Reflectance")
+
+for(i in 1:length(names(a))){
+h5writeAttribute(attr = a[[i]], h5obj=g, name=names(a[i]))
+}
+h5closeAll()
+```
+
+Next, we will loop through each of the datasets within the Coordinate_System group, and copy those (and their attributes, if present) from the full tile to our subset file. The Coordinate_System group contains many important pieces of information for geolocating our data, so we need to make sure that the sunset file has those data.
+
+```{r populate-group-attributes}
+
+# make a list of all groups within the full tile file
+ls=h5ls(f,all=T)
+
+#make a list of all of the names within the Coordinate_System group
+cg=unique(ls[ls$group=="/SJER/Reflectance/Metadata/Coordinate_System",]$name)
+
+for(i in 1:length(cg)){
+  print(cg[i])
+  a=h5readAttributes(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  d=h5read(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  attributes(d)=a
+  h5write(obj=d,file=nf,
+          name=paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]),
+          write.attributes=T)
+}
+```
+
+## Spectral Subsetting
+
+The goal of subsetting this dataset is to substantially reduce the file size to make it faster to download and process these data. While each AOP mosaic tile is not particularly large in terms of its spatial scale (1km by 1km at 1m resolution= 1,000,000 pixels, or about half as many pixels at shown on a standard 1080p computer screen), the 426 spectral bands available result in a fairly large file size. While having hundreds of bands is important for advanced research, we can use far fewer bands for demonstartion and teaching purposes, while still getting a good sense of what these hyperspectral data can do. Therefore, we will reduce the spectral resolution of these data by selecting every fourth band in the dataset, which reduces the file size to 1/4 of the original!
+
+```{r subset-wavelengths}
+#First, we make out 'index'  a list of number that will allow us to select every fourth band
+idx <- seq(from = 1, to = 426, by = 4)
+
+# We then use this index to select particular wavelengths from the full tile using the "index=" argument
+ws <- h5read(file = f, 
```

Comment:
why "ws" - not clear from the comment.

Reply:
shorthand for 'wavelengths', but I expanded to the full word in this revision to make it easier to read